testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -674,33 +692,244 @@ public void testOutputStats()
+                new OutputColumnMetadata("test_bigint", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "totalprice"))));
+    }
+
+    @Test
+    public void testOutputColumnsWithAggregationWithFilter()
+            throws Exception
+    {
+        assertColumnLineage(
+                "SELECT max(orderstatus) FILTER(WHERE orderdate > DATE '2000-01-01') as test_varchar, sum(totalprice) as test_bigint FROM orders GROUP BY custkey",
+                new OutputColumnMetadata(
+                        "test_varchar",
+                        ImmutableSet.of(
+                                new ColumnDetail("tpch", "tiny", "orders", "orderstatus"),
+                                new ColumnDetail("tpch", "tiny", "orders", "orderdate"))),
+                new OutputColumnMetadata("test_bigint", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "totalprice"))));
+    }
+
+    @Test
+    public void testOutputColumnsWithAggregationAndHaving()
+            throws Exception
+    {
+        assertColumnLineage(
+                "SELECT min(orderstatus) as test_varchar, sum(totalprice) as test_bigint FROM orders GROUP BY custkey HAVING min(orderdate) > DATE '2000-01-01'",
+                new OutputColumnMetadata("test_varchar", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "orderstatus"))),
+                new OutputColumnMetadata("test_bigint", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "totalprice"))));
+    }
+
+    @Test
+    public void testOutputColumnsWithCountAll()
+            throws Exception
+    {
+        assertColumnLineage(
+                "SELECT orderstatus as test_varchar, count(*) as test_bigint FROM orders GROUP BY orderstatus",
+                new OutputColumnMetadata("test_varchar", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "orderstatus"))),
+                new OutputColumnMetadata("test_bigint", ImmutableSet.of()));
+    }
+
+    @Test
+    public void testOutputColumnsWithWindowFunction()
+            throws Exception
+    {
+        assertColumnLineage(
+                "SELECT orderstatus as test_varchar, avg(totalprice) OVER (PARTITION BY custkey ORDER BY orderdate ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS test_bigint FROM orders",
+                new OutputColumnMetadata("test_varchar", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "orderstatus"))),
+                new OutputColumnMetadata(
+                        "test_bigint",
+                        ImmutableSet.of(
+                                new ColumnDetail("tpch", "tiny", "orders", "totalprice"),
+                                new ColumnDetail("tpch", "tiny", "orders", "custkey"),
+                                new ColumnDetail("tpch", "tiny", "orders", "orderdate"))));
+    }
+
+    @Test
+    public void testOutputColumnsWithPartialWindowClause()
```

Comment:
I'm not sure we want to differentiate between in-line window specification and WINDOW clause. Especially that those can be mixed within a single function call.
Currently, in-line window specification reports source columns, and WINDOW clause doesn't.

Reply:
This kind of a limitation for now. The long term plan is to report those columns too.